### DISCUSSION TITLE
How to upgrade bootc itself?

### DISCUSSION THREAD

**Discussion by Dakazan** [bootc-dev/bootc#641] [Q&A] [answered]
I've been using the `rhel-bootc:latest` image as a base to build container images via bootc-image-builder. I've recently wanted the feature for kernel args drop-in support, which was released for `bootc-0.1.12`. 

However, as of 2024/06/25, `rhel-bootc:latest` seems to contain `bootc-0.1.9`, which doesn't support this feature. (Which is another side topic, but I would expect `rhel-bootc:latest` to be tracking the bootc project with newer releases than 2 months ago...)

And it also seems the most recent version of bootc packaged through redhat repos is 0.1.9.
```bash
-> yum --showduplicates list available bootc
Available Packages
bootc.x86_64                                                                            0.1.7-1.el9                                                                              rhel-9-for-x86_64-appstream-rpms
bootc.x86_64                                                                            0.1.9-3.el9_4                                                                            rhel-9-for-x86_64-appstream-rpms
```

Sorry for all the redhat environment specific queries, just trying to give context.

Which brings the question, during my build stage, in the absence of recent rpms being packaged/available, what's the process to update bootc? 

Is it just, `make all` and then copy the new binary to `/usr/bin`? If so, are there any plans in place to distribute the rpms/binaries for each release on the releases page?

**Comment by miabbott** [comment]
RHEL moves slowly, so it might be a few weeks before the new version of `bootc` lands in the official `rhel-bootc` container images.

I think you are on the right track with `make all`; I would probably use a multi-stage container build to build the new binary in one stage and copy it into the resulting container image.

**Reply by Dakazan** [reply]
Awesome, thank you for the advice!

**Comment by cgwalters** [comment, marked as answer]
Hi, we are likely to ship bootc-0.1.12 even back to 9.4. But it's already queued in c9s (targeting 9.5) in https://gitlab.com/redhat/centos-stream/rpms/bootc/-/merge_requests/17 

For CentOS Stream there's an unpredictable latency for how long it takes to show up in the "production compose" i.e. what you get if you type "dnf update", or even how long it takes to appear in https://composes.stream.centos.org/development/

But it is in https://kojihub.stream.centos.org/koji/buildinfo?buildID=61611

We also have https://copr.fedorainfracloud.org/coprs/rhcontainerbot/bootc/ which...ok is broken, will look at that.

As for building from source, you can see our HACKING.md which touches on this a bit.

**Reply by Dakazan** [reply]
Thanks for the detailed response. I've ended up using the centos stream builds temporarily, and will either change to the multi-stage builder suggested earlier, or once the COPR repo builds are working I may even just use those.
